protocol/src/main/scala/io/mediachain/protocol/Datastore.scala

```diff
@@ -354,5 +359,19 @@ object Datastore {
       val optionals = Map("chain" -> chain.map(_.toCbor))
       super.toCMapWithDefaults(defaults, optionals)
     }
+
+    def toStream(datastore: Datastore): Streaming[JournalBlock] = {
+      val next: Eval[Streaming[JournalBlock]] = chain match {
+        case Some(ref) => Eval.always {
```

Comment:
right, I get what `Eval` generally does, but given that the two branches here are `.always` and a `.now` returning a constant, I'm assuming it's used as a trampoline for the recursion.